### ISSUE TITLE
Bypass NIfTI conversion

### ISSUE THREAD

**Issue by Lestropie** [PennLINC/ConFixel#25]
I couldn't help noticing from a quick perusal of your code that converting from fixel data `.mif` files to HDF5 currently requires an intermediate conversion to NIfTI-2 via *MRtrix3* `mrconvert`, such that the resulting `.nii` can be loaded using NiBabel.

It would be quite a lot more preferable if NiBabel could itself directly read `.mif` images.

I mentioned this in https://github.com/MRtrix3/mrtrix3/issues/2087, and it was also mentioned over at NiBabel in https://github.com/nipy/nibabel/issues/927. Unfortunately the intern I placed on the project wasn't able to finalise the contribution, but it might be possible to finish that code off and contribute it to NiBabel, which would likely be of use to many people but also would simplify the ConFixel code.

**Comment by mattcieslak** [comment]
@Lestropie that would be amazing! The nifti conversion is a big vulnerability here, as it sometimes flips the array of fixel values.

**Comment by Lestropie** [comment]
Yeah, that'll happen. Welcome to the wonderful world of [image strides](https://www.sciencedirect.com/science/article/pii/S1053811919307281#appsec1)... :grimacing:
